ddl/job_table.go

```diff
@@ -225,18 +227,35 @@ func (d *ddl) delivery2worker(wk *worker, pool *workerPool, job *model.Job) {
 			asyncNotify(d.ddlJobCh)
 			metrics.DDLRunningJobCount.WithLabelValues(pool.tp().String()).Dec()
 		}()
-		// we should wait 2 * d.lease time to guarantee all TiDB server have finished the schema change.
-		// see waitSchemaSynced for more details.
+		// check if this ddl job is synced to all servers.
 		if !d.isSynced(job) || d.once.Load() {
-			err := wk.waitSchemaSynced(d.ddlCtx, job, 2*d.lease)
```

Comment:
wk.waitSchemaSynced and waitSchemaChanged should provide worker retrun once found out the schema is not synced yet to avoid worker could not hanlder others DDL tasks.